mattermost/mattermost-plugin-mscalendar
# [MM-21140] Daily summary job

#### Summary

This PR allows a user to receive their daily summary of events at their configured time. The job is enabled/disabled using a system console variable. It runs every 30 minutes on the half-hour. UX around user configuration of their summary is explained in the QA steps.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21140

#### QA Steps

- Bot user must be connected with `/mscalendar connect_bot`
- Run `/mscalendar summary time 8:00AM` or whatever time in 30 minute increments that is coming up soon
- Run `/mscalendar summary enable` to enable your user's daily summary
- Enable the daily summary job in the system console
- Wait until the configured time comes up
- User should receive their daily summary of events
- If a 30 minute mark comes up and the user's time does not match, they should not receive a summary DM

#### Additional Info

Currently the time frame used to fetch the user's daily summary is `current time when summary is being sent + 24 hours`. This should be a more sophisticated query. [Ticket here](https://mattermost.atlassian.net/browse/MM-22649)

The operation takes into account the user's timezone that is set in Microsoft Calendar. We re-fetch their timezone whenever they set their configured time using `/mscalendar summary time 8:00AM` for example.
